k9mail/src/main/java/com/fsck/k9/activity/FolderInfoHolder.kt

```diff
@@ -0,0 +1,169 @@
+    }
+
+    override fun compareTo(other: FolderInfoHolder): Int {
+        val s1 = name ?: return -1
+        val s2 = other.name ?: return 1
+        val ret = s1.compareTo(s2, ignoreCase = true)
+        return if (ret != 0) {
+            ret
+        } else {
+            s1.compareTo(s2)
+        }
+    }
+
+    private fun truncateStatus(message: String?): String? {
+        var mess = message
+        if (mess != null && mess.length > 27) {
+            mess = mess.substring(0, 27)
+        }
+        return mess
+    }
+
+    // constructor for an empty object for comparisons
+    constructor()
+    constructor(context: Context?, folder: LocalFolder, account: Account) {
+        requireNotNull(context) { "null context given" }
+        populate(context, folder, account)
+    }
+
+    constructor(context: Context, folder: LocalFolder, account: Account, unreadCount: Int) {
+        populate(context, folder, account, unreadCount)
+    }
+
+    fun populate(context: Context, folder: LocalFolder, account: Account, unreadCount: Int) {
+        populate(context, folder, account)
+        unreadMessageCount = unreadCount
+        folder.close()
+    }
+
+    private fun populate(context: Context, folder: LocalFolder, account: Account) {
+        this.folder = folder
+        name = folder.name
+        lastChecked = folder.lastUpdate
+        status = truncateStatus(folder.status)
+        displayName = getDisplayName(context, account, name)
+        setMoreMessagesFromFolder(folder)
+    }
+
+    fun setMoreMessagesFromFolder(folder: LocalFolder) {
+        moreMessages = folder.hasMoreMessages()
+    }
+
+    companion object {
+        /**
+         * Returns the display name for a folder.
+         *
+         *
+         *
+         * This will return localized strings for special folders like the Inbox or the Trash folder.
+         *
+         *
+         * @param context
+         * A [Context] instance that is used to get the string resources.
+         * @param account
+         * The [Account] the folder belongs to.
+         * @param name
+         * The name of the folder for which to return the display name.
+         *
+         * @return The localized name for the provided folder if it's a special folder or the original
+         * folder name if it's a non-special folder.
+         */
+        @JvmStatic
+        fun getDisplayName(context: Context, account: Account, name: String?): String? {
+            return getDisplayName(context, account, name, name)
+        }
+
+        fun getDisplayName(
+            context: Context,
+            account: Account,
+            name: String?,
+            fqn: String?
+        ): String? {
+            // FIXME: We really shouldn't do a case-insensitive comparison here
+            return when {
+                fqn == account.spamFolderName ->
+                    String.format(
+                        context.getString(R.string.special_mailbox_name_spam_fmt), name
```

Comment:
that is good on the paper, but in reality of the actual project simply use K9.app.getString() for accessing resources. 
it is available while app is alive which can get rid of the NULL checks in constructor for the context which might disappear due to the activity/fragment lifecycle.